assets/src/edit-story/components/richText/fauxSelection.js

```diff
@@ -117,10 +117,14 @@ export function useFauxSelection(editorState, setEditorState) {
   }, [fauxSelection, editorState, setEditorState]);
 }
 
-export function fauxStylesToCSS(styles) {
+export function fauxStylesToCSS(styles, css) {
   const hasFauxSelection = styles.includes(FAUX_SELECTION);
   if (!hasFauxSelection) {
     return null;
   }
-  return { backgroundColor: 'rgba(169, 169, 169, 0.7)' };
+
+  return {
+    backgroundColor: 'rgba(169, 169, 169, 0.7)',
+    color: css?.color ? `var(--faux-selection-color, ${css.color})` : null,
```

Comment:
Yeah, this causes a problem, when you select something non-black (where it's a larger part that's non-black, not just the characters selected) and moves the focus out of the editor to change e.g. bold or italic in the design panel - the selected section becomes black. I assume it's because there's no `css.color` for just this subselection?

Demo:
![faux-wrong-color](https://user-images.githubusercontent.com/637548/100142378-c82c8b80-2e61-11eb-8a7c-5fdeb13eebe6.gif)

Reply:
I've added an update in ea9aa3e that seems to fix this specific issue, however.... further testing shows there is something wrong when selecting part of the text and editing the color the first time -- the selection is removed. To reproduce:

1. Mark some part of the text bold (e.g. as in your gif above)
2. Save & refresh
3. Go to edit mode of the text element and select the part that is bold
4. Change the color in the color-picker by dragging the pointer around -- see the selection gets removed!
5. Now select the same part again and try changing the color again -- works as expected.

Can you confirm you're seeing the same? Any idea why this might happen? You're probably more familiar with the internal life of the TextEditor. Also seeing this error throwing (console-logged it in `fauxSelection.js`):

![Screenshot 2020-11-25 at 11 24 57](https://user-images.githubusercontent.com/3294597/100208042-8e629000-2f08-11eb-87ac-4438ecffba76.png)

From the erroring file:
![Screenshot 2020-11-25 at 11 25 16](https://user-images.githubusercontent.com/3294597/100208084-9b7f7f00-2f08-11eb-9e7d-e5f7312ffa00.png)

I tested various options with assigning the CSS variable to see if I can detect what exactly causes this failure the first time but not the next times, but no luck so far -- any ideas?

Reply:
Hmph, odd, testing further it seems that just the `pushTransform` alone in `colorPicker` results in the selection getting lost the first time 😕

Reply:
OK, found the issue, should be fixed now.